shell/store/auth.js

```diff
@@ -130,10 +130,94 @@ export const actions = {
     commit('gotUser', user);
   },
 
+  async loggedIn({ dispatch, commit }) {
+    const me = await dispatch('findMe');
+
+    commit('hasAuth', true);
+    commit('loggedInAs', me.id);
+  },
+
+  notLoggedIn({ commit }) {
```

Comment:
I do not think we should handle redirects in the store.
The idea of the state management and actions is to commit changes to the store. If nothing is happening there, then it's either a helper or a service (if it contains its own variables) and it does not need to pass from the store.
The very fact that we pass the router logic as single actions and not value to a router logic set somewhere else makes it very complicated to find out where things happen.

Reply:
I also share similar concerns about handling routing from the store. Unfortunately, this is already a pattern that is pervasive throughout dashboard. We explicitly expose the router to the store for this very purpose

https://github.com/rancher/dashboard/blob/d8c0cf69bf1be6229ab7e58ac4b49dc3ab07d55d/shell/initialize/index.js#L46-L47

My initial instinct is that functions that handle dispatching actions and redirects  (`setProduct`, `invalidResource`, `handleFirstLogin`, etc..) would better fit as utility functions, but I can't say that's a better abstraction than what's represented here..

@codyrancher have we considered alternatives to using the store for these functions?

Reply:
I moved it to a utility.